datafusion/optimizer/src/optimize_projections/mod.rs

```diff
@@ -420,7 +436,12 @@ fn optimize_projections(
 
         optimize_projections(child, config, required_indices)?.transform_data(
             |new_input| {
-                if projection_beneficial {
+                if !config
+                    .options()
+                    .optimizer
+                    .optimize_projections_preserve_existing_projections
```

Comment:
Instead of a 4-line check, could we create a new function that takes a config object and returns true/false? It would make the code easier to read.

Reply:
As we've moved this check into `is_projection_unnecessary` and `add_projection_on_top_if_helpful` I think this is fine to access this flag directly instead of using additional helper function